click/_compat.py

```diff
@@ -482,9 +482,15 @@ def open_stream(filename, mode='r', encoding=None, errors='strict',
 
     # Non-atomic writes directly go out through the regular open functions.
     if not atomic:
-        if encoding is None:
-            return open(filename, mode), True
-        return io.open(filename, mode, encoding=encoding, errors=errors), True
+        open_kwargs = {}
+
+        if encoding is not None:
+            open_kwargs['encoding'] = encoding
+
+        if errors is not None:
+            open_kwargs['errors'] = errors
+
+        return codecs.open(filename, mode, **open_kwargs), True
```

Comment:
Looks like `codecs.open` has an early return that ignores `errors`, so we need to continue using `io.open`. I'll add back your check for `"b" not in mode`.